int16 x_offset
int16 y_offset